### ISSUE TITLE
File unexpectedly passes validation - should complain about arrays in response

### ISSUE THREAD

**Issue by jamescooke** [IBM/openapi-validator#71]
Am running v0.12.0 installed inside a Node 12.4 Docker container.

Given the following openapi v3 spec:

```
openapi: 3.0.0
info:
  description: Test arrays returned
  version: '0'
  title: Things service
servers:
  - url: 'https://example.com/test'
paths:
  '/things':
    post:
      summary: Do things
      operationId: do_things
      responses:
        '200':
          description: OK.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/ThingsArray'
components:
  schemas:
    ThingsArray:
      description: List of empty Things
      type: array
      items:
        type: object
```

This defines the response of `POST /things` to be an array of empty objects.

## Current behaviour

The file above passes the validator with all operations post, put, patch, delete, head, options, and trace - it only fails when 'get' is used.

It also passes when imaginary operations are used: look, shut, open.

## Expected behaviour

The file above should fail with:

```
errors

  Message :   Arrays MUST NOT be returned as the top-level structure in a response body.
  Path    :   paths./things.post.responses.200.content.application/json.schema
  Line    :   18

statistics

  Total number of errors   : 1
  Total number of warnings : 0

  errors
   1 (100%) : Arrays MUST NOT be returned as the top-level structure in a response body.
```

This should also happen for all other operations - real or created.

**Comment by dpopp07** [comment]
@jamescooke Thanks for the issue. This is certainly a bug and I will look into it this week.

**Comment by jamescooke** [comment]
@dpopp07 Thanks for confirming :+1: It's been driving me a bit crazy this afternoon.

**Comment by dpopp07** [comment]
Resolved in #73

**Comment by jamescooke** [comment]
@dpopp07 Thanks for the quick fix on this. Just tried with v0.12.2 and it's raising errors as expected :+1:
